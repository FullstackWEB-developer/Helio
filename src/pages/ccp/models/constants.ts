const contextPanels = {
    bot: 'bot',
    note: 'note',
    tickets: 'tickets',
    sms: 'sms',
    email: 'email',
    scripts: 'scripts'
}

export default contextPanels;