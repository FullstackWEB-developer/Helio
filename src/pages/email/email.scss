.email {
  &-sidebar {
    width: 22.5rem;
  }
  &-attachment-thumbnail {
    background-color: var(--table-header-background);
    &:hover {
      background-color: var(--light-divider-7);
    }
  }
  &-reply {
    max-height: 17.5rem;
    background-color: var(--light-divider-7);
    &-stripe {
      min-height: 3rem;
      background-color: var(--color-primary-default);
    }
  }
}

.collapsible-button {
  border: 1px solid var(--rounded-border-color);
}

.attachments-label {
  color: var(--color-primary-800);
}

.attachment-file-name {
  color: var(--light-text-high-emphasis);
}