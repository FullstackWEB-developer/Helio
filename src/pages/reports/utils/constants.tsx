import {ReportTypes} from '../models/report-types.enum'
import {ViewTypes} from '../models/view-types.enum'
import dayjs from 'dayjs';

export const viewTypes = [
    {
        label: 'reports.view_options.yesterday',
        value: ViewTypes.Yesterday.toString()
    },
    {
        label: 'reports.view_options.last_week',
        value: ViewTypes.LastWeek.toString()
    },
    {
        label: 'reports.view_options.last_7_days',
        value: ViewTypes.Last7Days.toString()
    },
    {
        label: 'reports.view_options.last_month',
        value: ViewTypes.LastMonth.toString()
    },
    {
        label: 'reports.view_options.monthly_reports',
        value: ViewTypes.MonthlyReports.toString()
    },
    {
        label: 'reports.view_options.custom_date_range',
        value: ViewTypes.CustomDates.toString()
    }
]

export const performanceChartViewTypes = [
    {
        label: 'reports.view_options.yesterday',
        value: ViewTypes.Yesterday.toString()
    },
    {
        label: 'reports.view_options.current_week',
        value: "7"
    },
    {
        label: 'reports.view_options.last_week',
        value: ViewTypes.LastWeek.toString()
    },
    {
        label: 'reports.view_options.current_month',
        value: "6"
    },
    {
        label: 'reports.view_options.last_month',
        value: ViewTypes.LastMonth.toString()
    }
];

export const reportTypes = [
    {
        label: 'reports.report_options.agent_reports',
        value: ReportTypes.AgentReports.toString()
    },
    {
        label: 'reports.report_options.bot_reports',
        value: ReportTypes.BotReports.toString()
    },
    {
        label: 'reports.report_options.system_reports',
        value: ReportTypes.SystemReports.toString()
    },
    {
        label: 'reports.report_options.queue_reports',
        value: ReportTypes.QueueReports.toString()
    }
]

const ONE_DAY_AS_SECOND = 86400;
const TWO_DAY_AS_SECOND = ONE_DAY_AS_SECOND * 2;
const TIME_FORMAT = 'HH:mm:ss'

export const getFormattedTime = (time, day, days) => {
    let innerText = "-";
    if (time) {
        let duration = dayjs.duration(time, 'seconds');
        if (time < ONE_DAY_AS_SECOND) {
            innerText = duration.format(TIME_FORMAT);
        } else if (time >= ONE_DAY_AS_SECOND && time < TWO_DAY_AS_SECOND) {
            innerText = `${duration.format('D')} ${day} ${duration.format(TIME_FORMAT)}`
        } else {
            innerText = `${duration.format('D')} ${days} ${duration.format(TIME_FORMAT)}`
        }
    }
    return innerText;
}
