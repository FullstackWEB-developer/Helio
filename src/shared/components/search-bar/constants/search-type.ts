export const searchType = {
    patientId: 7,
    patientName: 1,
    dateOfBirth: 3,
    ssn: 5,
    phone: 6
}