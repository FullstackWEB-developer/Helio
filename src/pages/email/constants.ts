import {PagedRequest} from '@shared/models';
import {getPageSize} from '@pages/contacts/contact-helpers/helpers';
import {EmailFilterModel} from '@pages/email/components/email-filter/email-filter.model';

export const NEW_EMAIL = 'new';

export const SystemUser = "System User";
export const Anonymous = "Anonymous";

export const DEFAULT_MESSAGE_QUERY_PARAMS: PagedRequest = {
    page: 1,
    pageSize: getPageSize()
}

export const DEFAULT_FILTER_VALUE: EmailFilterModel = {
    assignedTo: '',
    timePeriod: '',
}

export const DAYJS_LOCALE = {
    relativeTime: {
        future: "in %s",
        past: "%s ago",
        s: 'a few seconds',
        m: "1 min",
        mm: "%d mins",
        h: "1 h",
        hh: "%d h",
        d: "1 d",
        dd: "%d d",
        M: "1 mo",
        MM: "%d mo",
        y: "1 y",
        yy: "%d y"
    }
}
