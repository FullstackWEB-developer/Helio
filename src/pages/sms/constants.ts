import {getPageSize} from '@pages/contacts/contact-helpers/helpers';
import {PagedRequest} from '@shared/models';
import {SmsFilterParamModel} from './components/sms-filter/sms-filter.model';

export const DEFAULT_MESSAGE_QUERY_PARAMS: PagedRequest = {
    page: 1,
    pageSize: getPageSize()
}

export const DEFAULT_FILTER_VALUE: SmsFilterParamModel = {
    assignedTo: '',
    timePeriod: '1',
}

export const NEW_MESSAGE_OPTION_SELECT_TICKET = 'SELECT_TICKET';
export const NEW_MESSAGE_OPTION_CREATE_TICKET = 'CREATE_TICKET';
export const MORE_MENU_OPTION_PATIENT = 'VIEW_PATIENT';
export const MORE_MENU_OPTION_TICKET = 'VIEW_TICKET';
export const MORE_MENU_OPTION_CONTACT = 'VIEW_CONTACT';
