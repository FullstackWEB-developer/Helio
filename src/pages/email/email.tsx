import EmailLeftMenu from '@pages/email/components/email-left-menu';
import EmailConversation from '@pages/email/components/email-conversation/email-conversation';
import './email.scss'
import {getEnumByType, getLookupValues} from '@pages/tickets/services/tickets.service';
import {useEffect} from 'react';
import {useDispatch} from 'react-redux';
import {useParams} from 'react-router';
import {EMPTY_GUID} from '@pages/email/constants';
import NewEmail from '@pages/email/components/new-email/new-email';

const Email = () => {

    const dispatch = useDispatch();
    const {ticketId} = useParams<{ticketId: string}>();
    useEffect(() => {
        dispatch(getLookupValues('TicketReason'));
        dispatch(getEnumByType('TicketType'));
    }, [dispatch]);
    
    return <>
        <EmailLeftMenu/>
        {ticketId === EMPTY_GUID ? <NewEmail/>: <EmailConversation/>}
    </>
}

export default  Email;
