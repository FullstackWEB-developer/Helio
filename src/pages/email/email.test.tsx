import i18n from '../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import Email from './email';
import Router from "react-router-dom";
import EmailProvider from '@pages/email/context/email-context';
import {NEW_EMAIL} from '@pages/email/constants';
import MockDate from 'mockdate';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("Email tests", () => {
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        MockDate.set(dayjs('2018-04-04T16:00:00.000Z').toDate());
        container = document.createElement("div");
        document.body.appendChild(container);

    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
        MockDate.reset();
    });

    it("renders email correctly", async () => {
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <EmailProvider>
                <Email/>
            </EmailProvider>            
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders new-email with ticketId correctly", async () => {
        jest.spyOn(Router, 'useParams').mockReturnValue({ ticketId: NEW_EMAIL })
        jest.spyOn(Router, 'useParams').mockReturnValue({ ticketId: NEW_EMAIL })
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <EmailProvider>
                <Email/>
            </EmailProvider>            
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})
