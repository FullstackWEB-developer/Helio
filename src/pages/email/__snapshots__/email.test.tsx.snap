// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email tests renders email correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row email"
  >
    <div
      class="flex flex-col pt-6 border-r email-sidebar"
    >
      <div
        class="pb-2 pl-5 border-b"
      >
        <div
          class="relative"
        >
          <div
            class="relative flex flex-row items-center cursor-pointer flex-nowrap"
            data-testid="email-query-type-changer"
          >
            <div
              class="select-none whitespace-nowrap"
            >
              <h5>
                 email.query_type.my_email
              </h5>
            </div>
            <div
              class="px-3"
            >
              <div
                class=""
              >
                <svg
                  class="icon-medium "
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="24"
                    width="24"
                  />
                  <path
                    class="active-item-icon"
                    d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                    transform="translate(5 16.273) rotate(-90)"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="hidden"
        >
          <div>
            <div
              class="pl-5 flex py-4 border-b flex-col"
            >
              <div
                class="flex justify-between"
              >
                <div
                  class="subtitle"
                >
                  common.filters
                </div>
                <div
                  class="flex flex-row items-center"
                >
                  <div>
                    <div
                      class="pl-7 pr-4 cursor-pointer"
                    >
                      <svg
                        class="icon-small "
                        data-testid="svg-icon"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="fill-default"
                          d="M16,1.409,14.591,0,8,6.566,1.409,0,0,1.409,6.566,8,0,14.591,1.409,16,8,9.434,14.591,16,16,14.591,9.434,8Z"
                          transform="translate(4 4)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex flex-row pt-4"
              >
                <button
                  class="small-button cursor-pointer mr-4"
                  data-test-id="apply-button"
                  data-testid="apply-button"
                  type="button"
                >
                  <div
                    class="flex flex-row justify-center items-center space-x-2"
                  >
                    <div>
                      common.apply
                    </div>
                  </div>
                </button>
                <button
                  class="secondary-button cursor-pointer"
                  data-test-id="reset-all-button"
                  data-testid="reset-all-button"
                  type="button"
                >
                  <div
                    class="flex flex-row justify-center items-center space-x-2"
                  >
                    <div>
                      common.reset_all
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="flex flex-col pt-4 pl-5 pr-4"
            >
              <div>
                <div
                  class="cursor-pointer flex flex-row justify-between items-center h-10 mb-1.5"
                >
                  <div
                    class="subtitle"
                  >
                    email.filter.time_section
                  </div>
                  <div>
                    <div
                      class=""
                    >
                      <svg
                        class="cursor-pointer "
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="active-item-icon"
                          d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                          transform="translate(19 8) rotate(90)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <div
                      class="radio h-9 "
                    >
                      <input
                        data-test-id="timePeriod-radio"
                        data-testid="timePeriod_0"
                        id="timePeriod_0"
                        name="timePeriod"
                        type="radio"
                        value="0"
                      />
                      <label
                        class="body2  truncate "
                        for="timePeriod_0"
                      >
                        <div
                          class="truncate relative w-full h-full inline-block"
                        >
                          <span
                            class="body2"
                          >
                            common.time_periods.today
                          </span>
                        </div>
                      </label>
                    </div>
                    <div
                      class="radio h-9 "
                    >
                      <input
                        data-test-id="timePeriod-radio"
                        data-testid="timePeriod_1"
                        id="timePeriod_1"
                        name="timePeriod"
                        type="radio"
                        value="1"
                      />
                      <label
                        class="body2  truncate "
                        for="timePeriod_1"
                      >
                        <div
                          class="truncate relative w-full h-full inline-block"
                        >
                          <span
                            class="body2"
                          >
                            common.time_periods.last_7_days
                          </span>
                        </div>
                      </label>
                    </div>
                    <div
                      class="radio h-9 "
                    >
                      <input
                        data-test-id="timePeriod-radio"
                        data-testid="timePeriod_2"
                        id="timePeriod_2"
                        name="timePeriod"
                        type="radio"
                        value="2"
                      />
                      <label
                        class="body2  truncate "
                        for="timePeriod_2"
                      >
                        <div
                          class="truncate relative w-full h-full inline-block"
                        >
                          <span
                            class="body2"
                          >
                            common.time_periods.last_30_days
                          </span>
                        </div>
                      </label>
                    </div>
                    <div
                      class="radio h-9 "
                    >
                      <input
                        data-test-id="timePeriod-radio"
                        data-testid="timePeriod_3"
                        id="timePeriod_3"
                        name="timePeriod"
                        type="radio"
                        value="3"
                      />
                      <label
                        class="body2  truncate "
                        for="timePeriod_3"
                      >
                        <div
                          class="truncate relative w-full h-full inline-block"
                        >
                          <span
                            class="body2"
                          >
                            common.time_periods.date_range
                          </span>
                        </div>
                      </label>
                    </div>
                  </div>
                  <div
                    class="hidden"
                  >
                    <div
                      class="date-time-picker w-full flex flex-col h-20"
                    >
                      <div
                        class="date-time-picker-container flex flex-wrap relative"
                      >
                        <input
                          autocomplete="off"
                          class="relative truncate flex-shrink flex-grow flex-auto leading-normal w-px pl-4 body2 input-pt h-14 cursor-pointer"
                          data-testid="fromDate"
                          max="2018-04-04"
                          min="2017-10-04"
                          name="fromDate"
                          readonly=""
                          type="text"
                          value=""
                        />
                        <label
                          class="absolute truncate"
                        >
                          <span
                            class="label-span body2"
                          >
                            tickets.filter.from_date
                          </span>
                        </label>
                        <div
                          class="input-addon px-3 pt-4"
                          role="button"
                        >
                          <div
                            class=""
                          >
                            <svg
                              class="icon-medium "
                              height="24"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <rect
                                fill="none"
                                height="24"
                                width="24"
                              />
                              <path
                                class="date-time-picker-arrow"
                                d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                                transform="translate(5 16.273) rotate(-90)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="date-time-picker w-full flex flex-col h-20 date-time-picker-disabled"
                    >
                      <div
                        class="date-time-picker-container flex flex-wrap relative"
                      >
                        <input
                          autocomplete="off"
                          class="relative truncate flex-shrink flex-grow flex-auto leading-normal w-px pl-4 body2 input-pt h-14 cursor-pointer"
                          data-testid="toDate"
                          disabled=""
                          max="2018-04-04"
                          min="0001-01-01"
                          name="toDate"
                          readonly=""
                          type="text"
                          value=""
                        />
                        <label
                          class="absolute truncate"
                        >
                          <span
                            class="label-span body2"
                          >
                            tickets.filter.to_date
                          </span>
                        </label>
                        <div
                          class="input-addon px-3 pt-4"
                          role="button"
                        >
                          <div
                            class=""
                          >
                            <svg
                              class="icon-medium "
                              height="24"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <rect
                                fill="none"
                                height="24"
                                width="24"
                              />
                              <path
                                class="date-time-picker-arrow"
                                d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                                transform="translate(5 16.273) rotate(-90)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="flex-none border-b"
          >
            <div
              class="flex flex-row justify-between pt-4 pb-4 pl-5 pr-4"
            >
              <div
                class="subtitle2"
              />
              <div
                class="flex flex-row"
              >
                <div
                  class="mr-7"
                >
                  <svg
                    class="cursor-pointer icon-medium "
                    data-testid="new-email"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0,0H24V24H0Z"
                      fill="none"
                    />
                    <path
                      class="default-toolbar-icon"
                      d="M13.979,2.985l3.132,3.132a.34.34,0,0,1,0,.479L9.528,14.18l-3.222.358a.675.675,0,0,1-.747-.747l.358-3.222L13.5,2.985a.339.339,0,0,1,.479,0Zm5.625-.8L17.91.5a1.358,1.358,0,0,0-1.917,0L14.764,1.725a.34.34,0,0,0,0,.479L17.9,5.336a.339.339,0,0,0,.479,0L19.6,4.107a1.358,1.358,0,0,0,0-1.917Zm-6.271,9.927v3.535H2.222V4.541H10.2a.427.427,0,0,0,.3-.122l1.389-1.389a.417.417,0,0,0-.3-.712H1.667A1.667,1.667,0,0,0,0,3.985V16.208a1.667,1.667,0,0,0,1.667,1.667H13.889a1.667,1.667,0,0,0,1.667-1.667V10.728a.417.417,0,0,0-.712-.3l-1.389,1.389A.427.427,0,0,0,13.333,12.117Z"
                      transform="translate(2 2.9)"
                    />
                  </svg>
                </div>
                <div
                  class="relative"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="cursor-pointer icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="default-toolbar-icon"
                        d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="relative h-10 w-full search-input-field flex items-center h-12 flex-none"
          >
            <div
              class="absolute pl-4 pl-5"
            >
              <div
                class=""
              >
                <svg
                  class="cursor-pointer icon-small "
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="16"
                    width="16"
                  />
                  <path
                    class="search-icon-fill"
                    d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
                  />
                </svg>
              </div>
            </div>
            <input
              autocomplete="off"
              class="py-2.5 h-full w-full search-input-field body2 pl-12 border-b"
              placeholder="email.filter.search_placeholder"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="body3-small flex pt-4 px-6 text-center"
      >
        email.filter.no_emails
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Email tests renders new-email with ticketId correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row email"
  >
    <div
      class="flex flex-col pt-6 border-r email-sidebar"
    >
      <div
        class="pb-2 pl-5 border-b"
      >
        <div
          class="relative"
        >
          <div
            class="relative flex flex-row items-center cursor-pointer flex-nowrap"
            data-testid="email-query-type-changer"
          >
            <div
              class="select-none whitespace-nowrap"
            >
              <h5>
                 email.query_type.my_email
              </h5>
            </div>
            <div
              class="px-3"
            >
              <div
                class=""
              >
                <svg
                  class="icon-medium "
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="24"
                    width="24"
                  />
                  <path
                    class="active-item-icon"
                    d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                    transform="translate(5 16.273) rotate(-90)"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="hidden"
        >
          <div>
            <div
              class="pl-5 flex py-4 border-b flex-col"
            >
              <div
                class="flex justify-between"
              >
                <div
                  class="subtitle"
                >
                  common.filters
                </div>
                <div
                  class="flex flex-row items-center"
                >
                  <div>
                    <div
                      class="pl-7 pr-4 cursor-pointer"
                    >
                      <svg
                        class="icon-small "
                        data-testid="svg-icon"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="fill-default"
                          d="M16,1.409,14.591,0,8,6.566,1.409,0,0,1.409,6.566,8,0,14.591,1.409,16,8,9.434,14.591,16,16,14.591,9.434,8Z"
                          transform="translate(4 4)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex flex-row pt-4"
              >
                <button
                  class="small-button cursor-pointer mr-4"
                  data-test-id="apply-button"
                  data-testid="apply-button"
                  type="button"
                >
                  <div
                    class="flex flex-row justify-center items-center space-x-2"
                  >
                    <div>
                      common.apply
                    </div>
                  </div>
                </button>
                <button
                  class="secondary-button cursor-pointer"
                  data-test-id="reset-all-button"
                  data-testid="reset-all-button"
                  type="button"
                >
                  <div
                    class="flex flex-row justify-center items-center space-x-2"
                  >
                    <div>
                      common.reset_all
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="flex flex-col pt-4 pl-5 pr-4"
            >
              <div>
                <div
                  class="cursor-pointer flex flex-row justify-between items-center h-10 mb-1.5"
                >
                  <div
                    class="subtitle"
                  >
                    email.filter.time_section
                  </div>
                  <div>
                    <div
                      class=""
                    >
                      <svg
                        class="cursor-pointer "
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="active-item-icon"
                          d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                          transform="translate(19 8) rotate(90)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <div
                      class="radio h-9 "
                    >
                      <input
                        data-test-id="timePeriod-radio"
                        data-testid="timePeriod_0"
                        id="timePeriod_0"
                        name="timePeriod"
                        type="radio"
                        value="0"
                      />
                      <label
                        class="body2  truncate "
                        for="timePeriod_0"
                      >
                        <div
                          class="truncate relative w-full h-full inline-block"
                        >
                          <span
                            class="body2"
                          >
                            common.time_periods.today
                          </span>
                        </div>
                      </label>
                    </div>
                    <div
                      class="radio h-9 "
                    >
                      <input
                        data-test-id="timePeriod-radio"
                        data-testid="timePeriod_1"
                        id="timePeriod_1"
                        name="timePeriod"
                        type="radio"
                        value="1"
                      />
                      <label
                        class="body2  truncate "
                        for="timePeriod_1"
                      >
                        <div
                          class="truncate relative w-full h-full inline-block"
                        >
                          <span
                            class="body2"
                          >
                            common.time_periods.last_7_days
                          </span>
                        </div>
                      </label>
                    </div>
                    <div
                      class="radio h-9 "
                    >
                      <input
                        data-test-id="timePeriod-radio"
                        data-testid="timePeriod_2"
                        id="timePeriod_2"
                        name="timePeriod"
                        type="radio"
                        value="2"
                      />
                      <label
                        class="body2  truncate "
                        for="timePeriod_2"
                      >
                        <div
                          class="truncate relative w-full h-full inline-block"
                        >
                          <span
                            class="body2"
                          >
                            common.time_periods.last_30_days
                          </span>
                        </div>
                      </label>
                    </div>
                    <div
                      class="radio h-9 "
                    >
                      <input
                        data-test-id="timePeriod-radio"
                        data-testid="timePeriod_3"
                        id="timePeriod_3"
                        name="timePeriod"
                        type="radio"
                        value="3"
                      />
                      <label
                        class="body2  truncate "
                        for="timePeriod_3"
                      >
                        <div
                          class="truncate relative w-full h-full inline-block"
                        >
                          <span
                            class="body2"
                          >
                            common.time_periods.date_range
                          </span>
                        </div>
                      </label>
                    </div>
                  </div>
                  <div
                    class="hidden"
                  >
                    <div
                      class="date-time-picker w-full flex flex-col h-20"
                    >
                      <div
                        class="date-time-picker-container flex flex-wrap relative"
                      >
                        <input
                          autocomplete="off"
                          class="relative truncate flex-shrink flex-grow flex-auto leading-normal w-px pl-4 body2 input-pt h-14 cursor-pointer"
                          data-testid="fromDate"
                          max="2018-04-04"
                          min="2017-10-04"
                          name="fromDate"
                          readonly=""
                          type="text"
                          value=""
                        />
                        <label
                          class="absolute truncate"
                        >
                          <span
                            class="label-span body2"
                          >
                            tickets.filter.from_date
                          </span>
                        </label>
                        <div
                          class="input-addon px-3 pt-4"
                          role="button"
                        >
                          <div
                            class=""
                          >
                            <svg
                              class="icon-medium "
                              height="24"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <rect
                                fill="none"
                                height="24"
                                width="24"
                              />
                              <path
                                class="date-time-picker-arrow"
                                d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                                transform="translate(5 16.273) rotate(-90)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="date-time-picker w-full flex flex-col h-20 date-time-picker-disabled"
                    >
                      <div
                        class="date-time-picker-container flex flex-wrap relative"
                      >
                        <input
                          autocomplete="off"
                          class="relative truncate flex-shrink flex-grow flex-auto leading-normal w-px pl-4 body2 input-pt h-14 cursor-pointer"
                          data-testid="toDate"
                          disabled=""
                          max="2018-04-04"
                          min="0001-01-01"
                          name="toDate"
                          readonly=""
                          type="text"
                          value=""
                        />
                        <label
                          class="absolute truncate"
                        >
                          <span
                            class="label-span body2"
                          >
                            tickets.filter.to_date
                          </span>
                        </label>
                        <div
                          class="input-addon px-3 pt-4"
                          role="button"
                        >
                          <div
                            class=""
                          >
                            <svg
                              class="icon-medium "
                              height="24"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <rect
                                fill="none"
                                height="24"
                                width="24"
                              />
                              <path
                                class="date-time-picker-arrow"
                                d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                                transform="translate(5 16.273) rotate(-90)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="flex-none border-b"
          >
            <div
              class="flex flex-row justify-between pt-4 pb-4 pl-5 pr-4"
            >
              <div
                class="subtitle2"
              />
              <div
                class="flex flex-row"
              >
                <div
                  class="mr-7"
                >
                  <svg
                    class="cursor-pointer icon-medium "
                    data-testid="new-email"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0,0H24V24H0Z"
                      fill="none"
                    />
                    <path
                      class="default-toolbar-icon"
                      d="M13.979,2.985l3.132,3.132a.34.34,0,0,1,0,.479L9.528,14.18l-3.222.358a.675.675,0,0,1-.747-.747l.358-3.222L13.5,2.985a.339.339,0,0,1,.479,0Zm5.625-.8L17.91.5a1.358,1.358,0,0,0-1.917,0L14.764,1.725a.34.34,0,0,0,0,.479L17.9,5.336a.339.339,0,0,0,.479,0L19.6,4.107a1.358,1.358,0,0,0,0-1.917Zm-6.271,9.927v3.535H2.222V4.541H10.2a.427.427,0,0,0,.3-.122l1.389-1.389a.417.417,0,0,0-.3-.712H1.667A1.667,1.667,0,0,0,0,3.985V16.208a1.667,1.667,0,0,0,1.667,1.667H13.889a1.667,1.667,0,0,0,1.667-1.667V10.728a.417.417,0,0,0-.712-.3l-1.389,1.389A.427.427,0,0,0,13.333,12.117Z"
                      transform="translate(2 2.9)"
                    />
                  </svg>
                </div>
                <div
                  class="relative"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="cursor-pointer icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="default-toolbar-icon"
                        d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="relative h-10 w-full search-input-field flex items-center h-12 flex-none"
          >
            <div
              class="absolute pl-4 pl-5"
            >
              <div
                class=""
              >
                <svg
                  class="cursor-pointer icon-small "
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="16"
                    width="16"
                  />
                  <path
                    class="search-icon-fill"
                    d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
                  />
                </svg>
              </div>
            </div>
            <input
              autocomplete="off"
              class="py-2.5 h-full w-full search-input-field body2 pl-12 border-b"
              placeholder="email.filter.search_placeholder"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="body3-small flex pt-4 px-6 text-center"
      >
        email.filter.no_emails
      </div>
    </div>
  </div>
</DocumentFragment>
`;
